### PR TITLE
Update ITracer interface

### PR DESCRIPTION
**Pull request by andrewlock**
Reduces the number of overloads, and leaves the interface open for extension if required

@DataDog/apm-dotnet

### CODE REVIEW

File: tracer/src/Datadog.Trace/CreateSpanSettings.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// Settings to use when creating a new <see cref="IScope"/> and <see cref="ISpan"/>
+    /// </summary>
+    public struct CreateSpanSettings
```
Comment (pierotibou) [inline tracer/src/Datadog.Trace/CreateSpanSettings.cs:13]: I'm really bad at naming but I would have went with SpanCreationSettings

Comment (andrewlock) [reply tracer/src/Datadog.Trace/CreateSpanSettings.cs]: Done in https://github.com/DataDog/dd-trace-dotnet/pull/2131/commits/5a05d3d7c4f4bc9e05c1580758ec1585d3ed3c97

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2131 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710) will **not change** line coverage
:heavy_check_mark: Merging #2131 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710) will **not change** branch coverage
:no_entry: Merging #2131 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710) will will **increase** complexity by `2`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710) | #2131       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9105` / `12624` | `9090` / `12625` |          |
| Lines %   | `72%`      | `72%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4093` / `6117` | `4081` / `6119` |          |
| Branches %| `67%`      | `67%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7283`      | `7285`        |  `2`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=83048&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/83048/artifacts?artifactName=Code%20Coverage%20Report_83048&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2131](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=83116&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/83116/artifacts?artifactName=Code%20Coverage%20Report_83116&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710) | #2131       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `72%`      | `72%`       |  `0%` :heavy_check_mark:  |
| Branches %| `67%`      | `67%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7283`      | `7285`       |  `2` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/be7977887231c4fe4fade1ca9a86377b1ae08e2a/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/be7977887231c4fe4fade1ca9a86377b1ae08e2a/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/be7977887231c4fe4fade1ca9a86377b1ae08e2a/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-10%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/be7977887231c4fe4fade1ca9a86377b1ae08e2a/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=83048&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/83048/artifacts?artifactName=Code%20Coverage%20Report_83048&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2131](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=83116&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/83116/artifacts?artifactName=Code%20Coverage%20Report_83116&api-version=6.0&%24format=zip)

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2131 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710): 
* 1 benchmarks are **slower**, with geometric mean 1.142
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`WriteAndFlushEnrichedTraces`|net472|1,024.2&nbsp;μs|19.61&nbsp;μs|21.79&nbsp;μs|0.0000| 0.0000|0.0000|3.09&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|789.5&nbsp;μs|12.37&nbsp;μs|10.97&nbsp;μs|0.0000| 0.0000|0.0000|2.51&nbsp;KB|
|#2131|`WriteAndFlushEnrichedTraces`|net472|959.8&nbsp;μs|13.53&nbsp;μs|11.30&nbsp;μs|0.0000| 0.0000|0.0000|3.09&nbsp;KB|
|#2131|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|760.7&nbsp;μs|4.07&nbsp;μs|3.18&nbsp;μs|0.0000| 0.0000|0.0000|2.51&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Unknown :shrug: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`SendRequest`|net472|0.0000&nbsp;ns|0.0000&nbsp;ns|0.0000&nbsp;ns|0.0000| 0.0000|0.0000|0&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`SendRequest`|netcoreapp3.1|320,211.4905&nbsp;ns|5,758.3946&nbsp;ns|6,854.9591&nbsp;ns|0.1555| 0.0000|0.0000|20289&nbsp;B|
|#2131|`SendRequest`|net472|0.0000&nbsp;ns|0.0000&nbsp;ns|0.0000&nbsp;ns|0.0000| 0.0000|0.0000|0&nbsp;B|
|#2131|`SendRequest`|netcoreapp3.1|326,836.1798&nbsp;ns|4,772.5255&nbsp;ns|4,230.7195&nbsp;ns|0.1574| 0.0000|0.0000|20288&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`ExecuteNonQuery`|net472|339.3&nbsp;ns|5.83&nbsp;ns|5.46&nbsp;ns|0.0000| 0.0000|0.0000|0&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`ExecuteNonQuery`|netcoreapp3.1|303.4&nbsp;ns|4.10&nbsp;ns|3.84&nbsp;ns|0.0000| 0.0000|0.0000|0&nbsp;B|
|#2131|`ExecuteNonQuery`|net472|351.9&nbsp;ns|4.33&nbsp;ns|4.05&nbsp;ns|0.0000| 0.0000|0.0000|0&nbsp;B|
|#2131|`ExecuteNonQuery`|netcoreapp3.1|314.9&nbsp;ns|5.94&nbsp;ns|6.10&nbsp;ns|0.0000| 0.0000|0.0000|0&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #2131

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.ElasticsearchBenchmark.CallElasticsearch&#8209;net472 | 1.142 | 3,230.70 | 3,687.97 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`CallElasticsearch`|net472|3.247&nbsp;μs|0.0461&nbsp;μs|0.0431&nbsp;μs|0.1409| 0.0000|0.0000|907&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`CallElasticsearch`|netcoreapp3.1|1.828&nbsp;μs|0.0348&nbsp;μs|0.0372&nbsp;μs|0.0132| 0.0000|0.0000|928&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`CallElasticsearchAsync`|net472|3.635&nbsp;μs|0.0575&nbsp;μs|0.0538&nbsp;μs|0.1618| 0.0000|0.0000|1043&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`CallElasticsearchAsync`|netcoreapp3.1|1.902&nbsp;μs|0.0300&nbsp;μs|0.0266&nbsp;μs|0.0148| 0.0000|0.0000|1048&nbsp;B|
|#2131|`CallElasticsearch`|net472|3.698&nbsp;μs|0.0722&nbsp;μs|0.0914&nbsp;μs|0.1408| 0.0000|0.0000|907&nbsp;B|
|#2131|`CallElasticsearch`|netcoreapp3.1|1.966&nbsp;μs|0.0196&nbsp;μs|0.0173&nbsp;μs|0.0129| 0.0000|0.0000|928&nbsp;B|
|#2131|`CallElasticsearchAsync`|net472|3.996&nbsp;μs|0.0548&nbsp;μs|0.0486&nbsp;μs|0.1619| 0.0000|0.0000|1043&nbsp;B|
|#2131|`CallElasticsearchAsync`|netcoreapp3.1|2.003&nbsp;μs|0.0399&nbsp;μs|0.0908&nbsp;μs|0.0151| 0.0000|0.0000|1048&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`ExecuteAsync`|net472|3.954&nbsp;μs|0.0758&nbsp;μs|0.0779&nbsp;μs|0.1814| 0.0000|0.0000|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`ExecuteAsync`|netcoreapp3.1|2.258&nbsp;μs|0.0449&nbsp;μs|0.0965&nbsp;μs|0.0167| 0.0000|0.0000|1.14&nbsp;KB|
|#2131|`ExecuteAsync`|net472|3.921&nbsp;μs|0.0717&nbsp;μs|0.1988&nbsp;μs|0.1818| 0.0000|0.0000|1.14&nbsp;KB|
|#2131|`ExecuteAsync`|netcoreapp3.1|2.033&nbsp;μs|0.0262&nbsp;μs|0.0245&nbsp;μs|0.0161| 0.0000|0.0000|1.14&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`SendAsync`|net472|6.959&nbsp;μs|0.0833&nbsp;μs|0.0925&nbsp;μs|0.3599| 0.0000|0.0000|2.26&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`SendAsync`|netcoreapp3.1|4.546&nbsp;μs|0.0515&nbsp;μs|0.0482&nbsp;μs|0.0319| 0.0000|0.0000|2.25&nbsp;KB|
|#2131|`SendAsync`|net472|6.439&nbsp;μs|0.0983&nbsp;μs|0.0872&nbsp;μs|0.3633| 0.0000|0.0000|2.26&nbsp;KB|
|#2131|`SendAsync`|netcoreapp3.1|4.652&nbsp;μs|0.0613&nbsp;μs|0.0574&nbsp;μs|0.0323| 0.0000|0.0000|2.25&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|net472|4.611&nbsp;μs|0.0896&nbsp;μs|0.0880&nbsp;μs|0.3651| 0.0000|0.0000|2.27&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|netcoreapp3.1|4.530&nbsp;μs|0.0505&nbsp;μs|0.0472&nbsp;μs|0.0362| 0.0000|0.0000|2.58&nbsp;KB|
|#2131|`EnrichedLog`|net472|4.621&nbsp;μs|0.0613&nbsp;μs|0.0573&nbsp;μs|0.3646| 0.0000|0.0000|2.27&nbsp;KB|
|#2131|`EnrichedLog`|netcoreapp3.1|4.453&nbsp;μs|0.0554&nbsp;μs|0.0462&nbsp;μs|0.0378| 0.0000|0.0000|2.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|net472|183.6&nbsp;μs|1.68&nbsp;μs|1.57&nbsp;μs|0.7342| 0.2753|0.0000|5.43&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|netcoreapp3.1|162.1&nbsp;μs|1.92&nbsp;μs|1.80&nbsp;μs|0.0809| 0.0000|0.0000|5.42&nbsp;KB|
|#2131|`EnrichedLog`|net472|186.8&nbsp;μs|1.96&nbsp;μs|1.83&nbsp;μs|0.6659| 0.1903|0.0000|5.43&nbsp;KB|
|#2131|`EnrichedLog`|netcoreapp3.1|155.8&nbsp;μs|2.26&nbsp;μs|2.11&nbsp;μs|0.0762| 0.0000|0.0000|5.42&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|net472|12.33&nbsp;μs|0.246&nbsp;μs|0.230&nbsp;μs|0.8965| 0.0000|0.0000|5.58&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|netcoreapp3.1|11.32&nbsp;μs|0.152&nbsp;μs|0.142&nbsp;μs|0.1005| 0.0000|0.0000|6.92&nbsp;KB|
|#2131|`EnrichedLog`|net472|12.90&nbsp;μs|0.224&nbsp;μs|0.198&nbsp;μs|0.8981| 0.0000|0.0000|5.58&nbsp;KB|
|#2131|`EnrichedLog`|netcoreapp3.1|11.10&nbsp;μs|0.106&nbsp;μs|0.094&nbsp;μs|0.1002| 0.0000|0.0000|6.92&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`SendReceive`|net472|2.462&nbsp;μs|0.0405&nbsp;μs|0.0316&nbsp;μs|0.1669| 0.0000|0.0000|1.04&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`SendReceive`|netcoreapp3.1|2.223&nbsp;μs|0.0390&nbsp;μs|0.0365&nbsp;μs|0.0158| 0.0000|0.0000|1.12&nbsp;KB|
|#2131|`SendReceive`|net472|2.412&nbsp;μs|0.0271&nbsp;μs|0.0253&nbsp;μs|0.1671| 0.0000|0.0000|1.04&nbsp;KB|
|#2131|`SendReceive`|netcoreapp3.1|2.296&nbsp;μs|0.0437&nbsp;μs|0.1113&nbsp;μs|0.0160| 0.0000|0.0000|1.12&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|net472|9.173&nbsp;μs|0.0903&nbsp;μs|0.0800&nbsp;μs|0.5255| 0.0000|0.0000|3.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`EnrichedLog`|netcoreapp3.1|8.174&nbsp;μs|0.0900&nbsp;μs|0.0752&nbsp;μs|0.0452| 0.0000|0.0000|3.25&nbsp;KB|
|#2131|`EnrichedLog`|net472|9.340&nbsp;μs|0.1741&nbsp;μs|0.1543&nbsp;μs|0.5245| 0.0000|0.0000|3.28&nbsp;KB|
|#2131|`EnrichedLog`|netcoreapp3.1|8.257&nbsp;μs|0.1639&nbsp;μs|0.1610&nbsp;μs|0.0446| 0.0000|0.0000|3.25&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean |    Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`StartFinishSpan`|net472|1,039.9&nbsp;ns|20.23&nbsp;ns|45.24&nbsp;ns|0.0679| 0.0000|0.0000|433&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`StartFinishSpan`|netcoreapp3.1|922.9&nbsp;ns|13.89&nbsp;ns|12.32&nbsp;ns|0.0059| 0.0000|0.0000|432&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`StartFinishScope`|net472|1,266.7&nbsp;ns|22.10&nbsp;ns|35.68&nbsp;ns|0.0930| 0.0000|0.0000|594&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/91875901811272b2431f78204ca0aec179110710)|`StartFinishScope`|netcoreapp3.1|1,245.9&nbsp;ns|16.90&nbsp;ns|14.98&nbsp;ns|0.0100| 0.0000|0.0000|712&nbsp;B|
|#2131|`StartFinishSpan`|net472|963.5&nbsp;ns|11.15&nbsp;ns|9.88&nbsp;ns|0.0680| 0.0000|0.0000|433&nbsp;B|
|#2131|`StartFinishSpan`|netcoreapp3.1|919.9&nbsp;ns|10.56&nbsp;ns|9.88&nbsp;ns|0.0059| 0.0000|0.0000|432&nbsp;B|
|#2131|`StartFinishScope`|net472|1,236.5&nbsp;ns|14.65&nbsp;ns|13.70&nbsp;ns|0.0930| 0.0000|0.0000|594&nbsp;B|
|#2131|`StartFinishScope`|netcoreapp3.1|1,363.9&nbsp;ns|27.22&nbsp;ns|69.79&nbsp;ns|0.0098| 0.0000|0.0000|712&nbsp;B|
</details>

**Review by lucaspimentel — APPROVED** [review]

**Review by colin-higgins — APPROVED** [review]

**Review by pierotibou — APPROVED** [review]

**Pull request merged**